### PR TITLE
Add example for using multi-sheet Excel file

### LINKED ISSUE #1700: Document how to read and write a multi-sheet excel
## Introduction

>A high-level, short overview of the problem(s) you are designing a solution for.

We have the implementation, but it is not easy to point to a documented example:

image

## Background

> Provide the reader with the context surrounding the problem(s) you are trying to solve.

This is a common feature and something Kedro can already do.

### PR DESCRIPTION
**Pull request by ankatiyar**
## Description
Resolves #1700 

## Checklist

- [ ] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [ ] Added tests to cover my changes

<a href="https://gitpod.io/#https://github.com/kedro-org/kedro/pull/1760"></a>

### CODE REVIEW

File: docs/source/data/data_catalog.md [hunk 1]
```diff
@@ -164,7 +164,17 @@ rockets:
     sheet_name: Sheet1
 ```
 
-Example 7: Saves an image created with Matplotlib on Google Cloud Storage
+Example 7: Loads a multisheet excel file from a local file system
```
Comment (deepyaman) [inline docs/source/data/data_catalog.md:167]:
```suggestion
Example 7: Loads a multi-sheet Excel file from a local file system
```
:)

Comment (ankatiyar) [reply docs/source/data/data_catalog.md]: Made the change in 3f493a1921cf72bd3f81bd2f49f4ec3bedecc304, thanks!

**Review by noklam — APPROVED** [review]
Congrats on your first PR⭐️

Side notes: 
Follow up on previous discussion about DataSet API Docs should be the 1st entry point for people looking up examples, I think we should do the followings:

1. Remove these number, i.e. Example X, make them a sub-header like `## Save-an-image-xxxxx`, so we can link these example directly in the doc, and also better SEO I think?
2. Only keep some of the example in `Data Catalog` to showcase what Kedro can do, but leave the advance example in the Corresponding API Doc.
  * We have examples saving CSV on S3, matplotlib on GCP, excel on GCP, which I don't think add much value here, it's enough to show 1 example of GCP/S3.
  * Examples like saving HDF file doesn't worth the highlight in this page in my opinion, there could be more.

**Comment by datajoely** [comment]
Can I ask that we include the example how to save a multi-sheet as well? The instructions to do so are buried in in the DataSet docstring: https://github.com/kedro-org/kedro/blob/59064403e2590c992dceb51942f7dd9f6c97898a/kedro/extras/datasets/pandas/excel_dataset.py#L104

**Comment by MerelTheisenQB** [comment]
> Side notes: Follow up on previous discussion about DataSet API Docs should be the 1st entry point for people looking up examples, I think we should do the followings:
> 
> 1. Remove these number, i.e. Example X, make them a sub-header like `## Save-an-image-xxxxx`, so we can link these example directly in the doc, and also better SEO I think?
> 2. Only keep some of the example in `Data Catalog` to showcase what Kedro can do, but leave the advance example in the Corresponding API Doc.

Yes we should, but let's do that as separate tasks to keep this PR focussed on one thing. We've got this issue for task 1: https://github.com/kedro-org/kedro/issues/1742. I'll update it so it covers point 2 as well.

File: kedro/extras/datasets/pandas/excel_dataset.py [hunk 2]
```diff
@@ -65,6 +65,32 @@ class ExcelDataSet(
...
+
+    .. code-block:: yaml
+
+        >>> trains:
+        >>>   type: pandas.ExcelDataSet
+        >>>   filepath: data/02_intermediate/company/trains.xlsx
+        >>>   load_args:
+        >>>     sheet_name: [Sheet1, Sheet2, Sheet3]
+
+    Example multi-sheet excel file using Python API:
```
Comment (datajoely) [inline kedro/extras/datasets/pandas/excel_dataset.py:78]: So (as we discovered working this out together) the important piece of info that user needs to know is that to save a multi-sheet excel the user **doesn't** need to do anything fancy in `save_args`. Would you mind putting a comment explaining that.

Once again - really good work and thanks for the contribution!

Comment (ankatiyar) [reply kedro/extras/datasets/pandas/excel_dataset.py:82]: I've added the comment in f723f0de9e68e788f33f230ca4acc7121ad39219. Thank you!

File: kedro/extras/datasets/pandas/excel_dataset.py [hunk 3]
```diff
@@ -65,6 +65,36 @@ class ExcelDataSet(
         >>> reloaded = data_set.load()
         >>> assert data.equals(reloaded)
 
+    Note that using ``save_args`` is not required when saving multi-sheet excel files.
```
Comment (datajoely) [inline kedro/extras/datasets/pandas/excel_dataset.py:68]:
```suggestion
    Note: To save a multi-sheet excel files, no specials `save_args` are required.
```

File: kedro/extras/datasets/pandas/excel_dataset.py [hunk 4]
```diff
@@ -65,6 +65,36 @@ class ExcelDataSet(
...
+    A dict of ``pd.DataFrame`` is saved as a multi-sheet excel file by default.
```
Comment (datajoely) [inline kedro/extras/datasets/pandas/excel_dataset.py:69]:
```suggestion
    Instead, return a dictionary of ``Dict[str, pd.DataFrame]`` where the string keys are your sheet names.
```

**Review by datajoely — CHANGES_REQUESTED** [review]
Super close, slightly different wording suggested. Then I think we're good to go!

File: kedro/extras/datasets/pandas/excel_dataset.py (hunk 4 above)
Comment (ankatiyar) [reply kedro/extras/datasets/pandas/excel_dataset.py]: Done!

**Review by datajoely — APPROVED** [review]
🔥 awesome work @ankatiyar

**Review by yetudada — APPROVED** [review]
Congratulations on your first PR @ankatiyar! 🥳 And thank you for helping our users with this! We get this question a lot. 

I'm approving this but I just note that we use `excel` (lowercase) when it should be `Excel` (uppercase). This actually correct based on what we have in our documentation at the moment. 

Can you create a separate GitHub issue to address this as another quick-fix PR because I just checked the rest of the documentation and it looks like we've made that mistake in other places too?

**Pull request merged**